hummingbot/market/hitbtc/hitbtc_api_order_book_data_source.py

```diff
@@ -0,0 +1,299 @@
+            ]
+
+            old_symbols: List[str] = [
+                (
+                    f"{quoteAsset}-{baseAsset}"
+                )
+                for baseAsset, quoteAsset in zip(all_markets.baseAsset, all_markets.quoteAsset)
+            ]
+
+            all_markets.loc[:, "USDVolume"] = usd_volume
+            all_markets.loc[:, "old_symbol"] = old_symbols
+            await client.close()
+
+            return all_markets.sort_values("USDVolume", ascending=False)
+
+    async def get_trading_pairs(self) -> List[str]:
+        """
+        Return list of trading pairs
+        """
+        if not self._symbols:
+            try:
+                active_markets: pd.DataFrame = await self.get_active_exchange_markets()
+                self._symbols = active_markets.index.tolist()
+            except Exception:
+                self._symbols = []
+                self.logger().network(
+                    f"Error getting active exchange information.",
+                    exc_info=True,
+                    app_warning_msg=f"Error getting active exchange information. Check network connection.",
+                )
+
+        return self._symbols
+
+    @staticmethod
+    async def get_orderbook(trading_pair: str) -> Dict[str, any]:
+        """
+        Get whole orderbook
+        """
+        client = aiohttp.ClientSession()
+        orderbook_response = await client.get(f"{constants.REST_ORDERBOOK_URL}/{trading_pair}", params={"limit": 0})
+
+        if orderbook_response.status != 200:
+            raise IOError(
+                f"Error fetching OrderBook for {trading_pair} at {constants.EXCHANGE_NAME}. " f"HTTP status is {orderbook_response.status}."
+            )
+
+        orderbook_data: List[Dict[str, Any]] = await safe_gather(orderbook_response.json())
+
+        await client.close()
+
+        if len(orderbook_data) > 0:
+            return orderbook_data[0]
+
+        return {}
+
+    async def get_tracking_pairs(self) -> Dict[str, OrderBookTrackerEntry]:
+        trading_pairs: List[str] = await self.get_trading_pairs()
+        tracking_pairs: Dict[str, OrderBookTrackerEntry] = {}
+
+        number_of_pairs: int = len(trading_pairs)
+        for index, trading_pair in enumerate(trading_pairs):
+            try:
+                snapshot: Dict[str, any] = await self.get_orderbook(trading_pair)
+                snapshot_timestamp: float = pd.Timestamp(snapshot["timestamp"]).timestamp()
+                snapshot_msg: OrderBookMessage = HitbtcOrderBook.snapshot_message_from_exchange(
+                    snapshot,
+                    snapshot_timestamp,
+                    metadata={"trading_pair": trading_pair}
+                )
+                order_book: OrderBook = self.order_book_create_function()
+                active_order_tracker: HitbtcActiveOrderTracker = HitbtcActiveOrderTracker()
+                bids, asks = active_order_tracker.convert_snapshot_message_to_order_book_row(snapshot_msg)
```

Comment:
It seems there is an error with the `convert_snapshot_message_to_order_book_row()` function. 
Since this is failing, the order book is not initialized and the tests is stuck waiting for the snapshot message to be processed and applied to the order book

Reply:
I think I forgot to change the snapshot conversion in accordance with API req.